servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAlivePolicies.java

```diff
@@ -0,0 +1,143 @@
+     */
+    public static KeepAlivePolicy whenIdleFor(final Duration idleDuration, final Duration ackTimeout) {
+        return new KeepAlivePolicyBuilder().idleDuration(idleDuration).ackTimeout(ackTimeout).build();
+    }
+
+    /**
+     * A builder of {@link KeepAlivePolicy}.
+     */
+    public static final class KeepAlivePolicyBuilder {
+        private Duration idleDuration = DEFAULT_IDLE_DURATION;
+        private Duration ackTimeout = DEFAULT_ACK_TIMEOUT;
+        private boolean withoutActiveStreams;
+
+        /**
+         * Set the {@link Duration} of idleness on a connection after which a
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+         * <p>
+         * <strong>Too short ping durations may cause high network traffic, implementations may enforce a minimum
+         * duration.</strong>
+         *
+         * @param idleDuration {@link Duration} of idleness on a connection after which a
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+         * @return {@code this}.
+         * @see KeepAlivePolicy#idleDuration()
+         */
+        public KeepAlivePolicyBuilder idleDuration(final Duration idleDuration) {
+            if (idleDuration.getSeconds() < 10) {
+                throw new IllegalArgumentException("idleDuration: " + idleDuration + " (expected >= 10 seconds");
+            }
+            this.idleDuration = idleDuration;
+            return this;
+        }
+
+        /**
+         * Set the maximum {@link Duration} to wait for an acknowledgment of a previously sent
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>. If no acknowledgment is received, the
+         * connection will be closed.
+         *
+         * @param ackTimeout {@link Duration} to wait for an acknowledgment of a previously sent
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>.
+         * @return {@code this}.
+         * @see KeepAlivePolicy#ackTimeout()
+         */
+        public KeepAlivePolicyBuilder ackTimeout(final Duration ackTimeout) {
+            this.ackTimeout = requireNonNull(ackTimeout);
+            return this;
+        }
+
+        /**
+         * Allow/disallow sending or receiving <a href="https://tools.ietf.org/html/rfc7540#section-6.7">pings</a> even
+         * when no streams are <a href="https://tools.ietf.org/html/rfc7540#section-5.1">active</a>.
+         *
+         * @param withoutActiveStreams {@code true} if
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">pings</a> are expected when no streams are
+         * <a href="https://tools.ietf.org/html/rfc7540#section-5.1">active</a>.
+         * @return {@code this}.
+         * @see KeepAlivePolicy#withoutActiveStreams()
+         */
+        public KeepAlivePolicyBuilder withoutActiveStreams(final boolean withoutActiveStreams) {
```

Comment:
WDYT about `whenNoActiveStreams` name here and for `KeepAlivePolicy`? "without" does not match with the javadoc language.

Reply:
I guess they are equivalent:

`keepAlive.withoutActiveStreams()` or `keepAlive.withNoActiveStreams()`

`without` is a single negative whereas `withNo` is a positive (with) and a negative (no), so I feel `without` is easier to understand. 

Javadocs language has more context than a name so I feel it is OK to be different.